portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -133,20 +132,24 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
    */
   @Override
   public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the request
-    String idString = request.getParameter("comment-id");
+    // Make sure the user deleting comments is an admin
+    UserService userService = UserServiceFactory.getUserService();
+    if(userService.isUserAdmin()) {
```

Comment:
Prefer to follow the principle of the least indentation. Compare:

```
void func() {
  if(blah) {
    // lots of code
  }
}
```

vs

```
void func() {
  if(!blah) {
    return;
  }
  // lots of code
}
```

In the second example, 'lots of code' has less indentation, and it is potentially easier to read. Also, by having an early exit from the function it signals to whoever reads your code that once that condition is satisfied (i.e. in your case it would be when a user is not an admin) the rest of the function can be safely ingored.

Reply:
Indeed, I didn't know about this principle, but it makes a lot of sense!

Thank you